server/connector.js

```diff
@@ -201,8 +201,12 @@ connector._loadFromFile = function(cb) {
           PackageDefinition.addToCache(cache, f.data);
           if (dir === '.') {
             var loopBackVersion = f.data.dependencies['loopback'] ||
-              f.data.devDependencies['loopback'] ||
-              f.data.optionalDependencies['loopback'];
+              f.data.devDependencies['loopback'];
```

Comment:
There will be similar situations where `dependencies` or `devDependencies` is undefined.